server/client/conn.go

```diff
@@ -415,9 +415,9 @@ func (c *Conn) allocateMessageId(f *frame.Frame, sub *Subscription) {
 		// if there is any requirement by the client to acknowledge, set
 		// the ack header as per STOMP 1.2
 		if sub == nil || sub.ack == frame.AckAuto {
-			f.Header.Del(frame.Ack)
+			f.Header.Del(frame.Id)
 		} else {
-			f.Header.Set(frame.Ack, messageId)
+			f.Header.Set(frame.Id, messageId)
```

Comment:
@Delorus @worg 

It is my understanding that only `ACK` frames should use the `id` header. This `MESSAGE` frame should still use `ack` header. https://stomp.github.io/stomp-specification-1.2.html#MESSAGE

Or am I mistaken ?

Reply:
@dselim you're right, sorry my bad 😔

Reply:
no worries!